opnode/eth/id.go

```diff
@@ -20,3 +20,34 @@ func (id BlockID) String() string {
 func (id BlockID) TerminalString() string {
 	return fmt.Sprintf("%s:%d", id.Hash.TerminalString(), id.Number)
 }
+
+type L2Node struct {
```

Comment:
Do you have a suggestion here? @norswap also thinks that is should change.
I don't love `Reference` as it tended to become really repetitive (I saw a lot of `xxRef` variable names, and it felt like it lost meaning to me), though I wonder if that could be mitigated and using `Ref` in the type name would be ok.
Otherwise, I like `node` because it is evocative of a graph. I think `block`/`header` would work similar (though we do strip out info).


Some potential names (feel free to suggest more)
- `BlockRef`
- `BlockReference`
- `Link`
- `Block`
- `Header`
- `Vertex`

Reply:
`L1Ref` and `L2Ref` would work imho. This avoids confusion with the node software / main object, and is simple and short. All those variable names are using "ref" because of the same meaning (always multiple vars that reference L1, L2 etc all together). We can probably refactor more code to use these `Ref` structs and reduce variables too.